test/hack/bin/verify-rules.sh

```diff
@@ -0,0 +1,134 @@
+    local HELM=test/hack/bin/helm
+    local ARCHITECT=test/hack/bin/architect
+    local YQ=test/hack/bin/yq
+
+    # prepare the helm chart
+    echo "### Helm chart preparation"
+    "$GIT_WORKDIR/$ARCHITECT" helm template --dir "$GIT_WORKDIR/helm/prometheus-rules" --dry-run
+
+    expected_failure_relative_file="test/conf/promtool_ignore"
+    expected_failure_file="$GIT_WORKDIR/$expected_failure_relative_file"
+
+    # Retrieve all files we're going to check
+    local -a all_files
+    mapfile -t all_files < <(
+        cd "$GIT_WORKDIR" || return 1
+        # filter alerting-rules files, and remove prefix `helm/prometheus-rules/`
+        git ls-files |
+            sed -En 's_^helm/prometheus-rules/(templates/alerting-rules/.*\.ya?ml)$_\1_p' || echo error
+    )
+
+    # Get prefixes whitelisted via the failure_file
+    local -a expected_failure_prefixes
+    mapfile -t expected_failure_prefixes <"$expected_failure_file"
+
+    local -a providers
+    mapfile -t providers <"$GIT_WORKDIR/test/conf/providers"
+
+    local -a promtool_check_errors=()
+    local -a promtool_test_errors=()
+
+    for file in "${all_files[@]}"; do
+
+        # if the $file is whitelisted via the $expected_failure_file
+        if array_contains "$file" "${expected_failure_prefixes[@]}"; then
+            # don't run tests for it
+            echo "### Skipping $file"
+            continue
+        fi
+
+        echo "### Testing $file"
+        for provider in "${providers[@]}"; do
+            echo "###    Provider: $provider"
+
+            # retrieve basename in pure bash
+            filename="${file##*/}"
+
+            # don't run tests if no provider specific tests are defined
+            if [[ ! -d "$GIT_WORKDIR/test/tests/providers/$provider" ]]; then
+                echo "###   No tests for proviter $provider - skipping"
+                continue
+            fi
+
+            echo "###    extracting $GIT_WORKDIR/test/providers/$provider/$filename"
+            "$GIT_WORKDIR/$HELM" template \
+                --set="managementCluster.provider.kind=$provider" \
+                --release-name prometheus-rules \
+                --namespace giantswarm "$GIT_WORKDIR"/helm/prometheus-rules \
+                -s "$file" |
+                $YQ '.spec' - >"$GIT_WORKDIR/test/tests/providers/$provider/$filename"
+
+            echo "###    promtool check rules $GIT_WORKDIR/test/tests/providers/$provider/$filename"
+            local promtool_check_output
+            promtool_check_output="$("$GIT_WORKDIR/$PROMTOOL" check rules "$GIT_WORKDIR/test/tests/providers/$provider/$filename" 2>&1)" ||
+                promtool_check_errors+=("$promtool_check_output")
+
+            local promtool_test_output
+            local testfile="$GIT_WORKDIR/test/tests/providers/$provider/${filename%.yml}.test.yml"
+            if [[ ! -f "$testfile" ]]; then
+                echo "###    testfile $testfile not found, skipping promtool test"
```

Comment:
Bug: It says it skips, then it does not.
```suggestion
                echo "###    testfile $testfile not found, skipping promtool test"
                continue
```